File: Components/WinForms/VB/EditorsMainDemo/Modules/ModuleProgressPanel.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Utils.Animation
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors

Namespace DevExpress.XtraEditors.Demos
    Partial Public Class ModuleProgressPanel
        Inherits TutorialControl

        Public Sub New()
            InitializeComponent()
            TutorialInfo.WhatsThisCodeFile = "VB\EditorsMainDemo\Modules\ModuleProgressPanel.vb"
            TutorialInfo.WhatsThisXMLFile = "progressPanel.xml"
        End Sub
        ' <comboBoxAnimationType>
        Private Sub comboBoxAnimationType_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxAnimationType.SelectedIndexChanged
            Select Case CStr(comboBoxAnimationType.SelectedItem)
                Case ("Default")
                    progressPanelSample.WaitAnimationType = WaitingAnimatorType.Default
                Case ("Line")
                    progressPanelSample.WaitAnimationType = WaitingAnimatorType.Line
                Case ("Ring")
                    progressPanelSample.WaitAnimationType = WaitingAnimatorType.Ring
            End Select
            ' <skip>
            spinElementCount.Enabled = (progressPanelSample.WaitAnimationType <> WaitingAnimatorType.Default)
            spinAcceleration.Enabled = (progressPanelSample.WaitAnimationType <> WaitingAnimatorType.Default)
            spinSpeed.Enabled = (progressPanelSample.WaitAnimationType <> WaitingAnimatorType.Default)
            spinLineHeight.Enabled = (progressPanelSample.WaitAnimationType = WaitingAnimatorType.Line)
            comboBoxLineElementType.Enabled = (progressPanelSample.WaitAnimationType = WaitingAnimatorType.Line)
            spinRingDiameter.Enabled = (progressPanelSample.WaitAnimationType = WaitingAnimatorType.Ring)
            ' </skip>
        End Sub
        ' </comboBoxAnimationType>

        ' <spinElementCount>
        Private Sub spinElementCount_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinElementCount.EditValueChanged
            progressPanelSample.AnimationElementCount = CInt((spinElementCount.Value))
        End Sub
        ' </spinElementCount>

        ' <spinAcceleration>
        Private Sub spinAcceleration_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinAcceleration.EditValueChanged
            progressPanelSample.AnimationAcceleration = CSng(spinAcceleration.Value)
        End Sub
        ' </spinAcceleration>

        ' <spinSpeed>
        Private Sub spinSpeed_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinSpeed.EditValueChanged
            progressPanelSample.AnimationSpeed = CSng(spinSpeed.Value)
        End Sub
        ' </spinSpeed>

        ' <spinLineHeight>
        Private Sub spinLineHeight_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinLineHeight.EditValueChanged
            progressPanelSample.LineAnimationElementHeight = CInt((spinLineHeight.Value))
        End Sub
        ' </spinLineHeight>

        ' <comboBoxLineElementType>
        Private Sub comboBoxLineElementType_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxLineElementType.SelectedIndexChanged
            Select Case CStr(comboBoxLineElementType.SelectedItem)
                Case ("Circle")
                    progressPanelSample.LineAnimationElementType = LineAnimationElementType.Circle
                Case ("Rectangle")
                    progressPanelSample.LineAnimationElementType = LineAnimationElementType.Rectangle
                Case ("Triangle")
                    progressPanelSample.LineAnimationElementType = LineAnimationElementType.Triangle
            End Select
        End Sub
        ' </comboBoxLineElementType>

        ' <spinRingDiameter>
        Private Sub spinRingDiameter_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinRingDiameter.EditValueChanged
            progressPanelSample.RingAnimationDiameter = CInt((spinRingDiameter.Value))
        End Sub
        ' </spinRingDiameter>

        ' <colorEditElementColor>
        Private Sub colorEditElementColor_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles colorEditElementColor.EditValueChanged
            progressPanelSample.Appearance.ForeColor = colorEditElementColor.Color
        End Sub
        ' </colorEditElementColor>

        ' <spinFrameCount>
        Private Sub spinFrameCount_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinFrameCount.EditValueChanged
            progressPanelSample.FrameCount = CInt((spinFrameCount.Value))
        End Sub
        ' </spinFrameCount>

        ' <spinFrameInterval>
        Private Sub spinFrameInterval_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinFrameInterval.EditValueChanged
            progressPanelSample.FrameInterval = CInt((spinFrameInterval.Value))
        End Sub
        ' </spinFrameInterval>
    End Class
End Namespace

File: Components/WinForms/CS/EditorsMainDemo/Modules/ModuleProgressPanel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils.Animation;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors;

namespace DevExpress.XtraEditors.Demos {
    public partial class ModuleProgressPanel : TutorialControl {
        public ModuleProgressPanel() {
            InitializeComponent();
            TutorialInfo.WhatsThisCodeFile = "CS\\EditorsMainDemo\\Modules\\ModuleProgressPanel.cs";
            TutorialInfo.WhatsThisXMLFile = "DevExpress.XtraEditors.Demos.CodeInfo.progressPanel.xml";
        }

        protected override void UpdateTabControlPosition(XtraTab.XtraTabControl tabControl) { }
        protected override void CalcContentCore() {
            CalcLayoutControlBounds(layoutControl1);
        }
        // <comboBoxAnimationType>
        private void comboBoxAnimationType_SelectedIndexChanged(object sender, EventArgs e) {
            switch((string)comboBoxAnimationType.SelectedItem) {
                case ("Default"):
                    progressPanelSample.WaitAnimationType = WaitingAnimatorType.Default;
                    break;
                case ("Line"):
                    progressPanelSample.WaitAnimationType = WaitingAnimatorType.Line;
                    break;
                case ("Ring"):
                    progressPanelSample.WaitAnimationType = WaitingAnimatorType.Ring;
                    break;
            }
            // <skip>
            spinElementCount.Enabled = (progressPanelSample.WaitAnimationType != WaitingAnimatorType.Default);
            spinAcceleration.Enabled = (progressPanelSample.WaitAnimationType != WaitingAnimatorType.Default);
            spinSpeed.Enabled = (progressPanelSample.WaitAnimationType != WaitingAnimatorType.Default);
            colorEditElementColor.Enabled = (progressPanelSample.WaitAnimationType != WaitingAnimatorType.Default);
            spinLineHeight.Enabled = (progressPanelSample.WaitAnimationType == WaitingAnimatorType.Line);
            comboBoxLineElementType.Enabled = (progressPanelSample.WaitAnimationType == WaitingAnimatorType.Line);
            spinRingDiameter.Enabled = (progressPanelSample.WaitAnimationType == WaitingAnimatorType.Ring);
            // </skip>
        }
        // </comboBoxAnimationType>

        // <spinElementCount>
        private void spinElementCount_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.AnimationElementCount = (int)spinElementCount.Value;
        }
        // </spinElementCount>

        // <spinAcceleration>
        private void spinAcceleration_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.AnimationAcceleration = (float)spinAcceleration.Value;
        }
        // </spinAcceleration>

        // <spinSpeed>
        private void spinSpeed_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.AnimationSpeed = (float)spinSpeed.Value;
        }
        // </spinSpeed>

        // <spinLineHeight>
        private void spinLineHeight_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.LineAnimationElementHeight = (int)spinLineHeight.Value;
        }
        // </spinLineHeight>

        // <comboBoxLineElementType>
        private void comboBoxLineElementType_SelectedIndexChanged(object sender, EventArgs e) {
            switch((string)comboBoxLineElementType.SelectedItem) {
                case ("Circle"):
                    progressPanelSample.LineAnimationElementType = LineAnimationElementType.Circle;
                    break;
                case ("Rectangle"):
                    progressPanelSample.LineAnimationElementType = LineAnimationElementType.Rectangle;
                    break;
                case ("Triangle"):
                    progressPanelSample.LineAnimationElementType = LineAnimationElementType.Triangle;
                    break;
            }
        }
        // </comboBoxLineElementType>

        // <spinRingDiameter>
        private void spinRingDiameter_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.RingAnimationDiameter = (int)spinRingDiameter.Value;
        }
        // </spinRingDiameter>

        // <colorEditElementColor>
        private void colorEditElementColor_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.Appearance.ForeColor = colorEditElementColor.Color;
        }
        // </colorEditElementColor>

        // <spinFrameCount>
        private void spinFrameCount_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.FrameCount = (int)spinFrameCount.Value;
        }
        // </spinFrameCount>

        // <spinFrameInterval>
        private void spinFrameInterval_EditValueChanged(object sender, EventArgs e) {
            progressPanelSample.FrameInterval = (int)spinFrameInterval.Value;
        }
        // </spinFrameInterval>
    }
}
